docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+by publishing a message to the topic `te/device/main///cmd/log_upload/<op_id>`.
+
+So, for example, the operation status update message should look like below. The topic is `te/device/main///cmd/log_upload/1234`.
+
+```json
+// Why do we need to repeat other fileds than status and reason?
+{
+  "status": "failed",
+  "reason": "The target log file for 'mosquitto' does not exist."
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000,
+  "reason": "The type 'typeA' is not configured."
+}
+```
+
+### Successful flow
+
+The status transition: `init` -> `executing` -> `uploading` -> `successful`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP]
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP]
+  alt if necessary
+    Mapper->>Cloud: Status: successful
+    Mapper->>Cloud: Cloud-specific something if any
+  end
+```
+
+### Failed flows
+
+a. Failure on the plugin side
+
+The status transition: `init` -> `executing` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log (FAILED!)
+  Plugin->>Mapper: Status: failed with reason
+  alt if necessary
+    Mapper->>Cloud: Status: failed with reason
+  end
+```
+
+b. Failure on uploading log (mapper side)
+
+The status transition: `init` -> `executing` -> `uploading` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP] (FAILED!)
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP] (FAILED CODE!)
```

Comment:
I didn't really understand why the cloud would return a URL when the upload failed.

Reply:
It's a copy & paste mistake!